models/Product.php

```diff
@@ -228,6 +253,22 @@ public function afterSave($insert, $changedAttributes)
             $this->linkAll('categories', $categories);
         }
 
+        if ($this->vouchers) {
+            foreach (array_unique(array_map('trim', array_filter(explode(";", $this->vouchers)))) as $voucherValue) {
```

Comment:
put array_filter around array_map, not inside.. like in case of a space: `a; ;b`, but it's not code breaking

Reply:
nice catch , thanks , done